types/signed_header.go

```diff
@@ -118,16 +150,20 @@ func (sh *SignedHeader) ValidateBasic() error {
 	}
 
 	signature := sh.Commit.Signatures[0]
-	proposer := sh.Validators.GetProposer()
-	var pubKey ed25519.PubKey = proposer.PubKey.Bytes()
+
 	msg, err := sh.Header.MarshalBinary()
 	if err != nil {
-		return errors.New("signature verification failed, unable to marshal header")
+		return fmt.Errorf("signature verification failed, unable to marshal header: %v", err)
+	}
+	if sh.Validators.Validators[0].PubKey == nil {
+		return errors.New("public key in validator is nil")
 	}
-	if !pubKey.VerifySignature(msg, signature) {
+	if signature == nil {
```

Comment:
no need to check this either. `VerifySignature` fails if nil or not appropriate size